src/transformers/trainer_pt_utils.py

```diff
@@ -55,6 +55,14 @@
 logger = logging.get_logger(__name__)
 
 
+def check_dataloader_randomsampler(dataloader):
+    if hasattr(dataloader, "sampler") and isinstance(dataloader.sampler, RandomSampler):
+        return dataloader.sampler, True
+    if hasattr(dataloader, "batch_sampler"):
+        return check_dataloader_randomsampler(dataloader.batch_sampler)
+    return dataloader.sampler, False
```

Comment:
This should just return the sampler, and then the user can choose what they do with the output e.g. check if it's a random sampler. This ensures the function is as versatile as possible and can be used / extended without issue. 


```suggestion
def get_dataloader_sampler(dataloader):
    if hasattr(dataloader, "sampler"):
        return dataloader.sampler
    if hasattr(dataloader, "batch_sampler"):
        return get_dataloader_sampler(dataloader.batch_sampler)
```